subprojects/platform-jvm/src/main/java/org/gradle/jvm/tasks/Jar.java

```diff
@@ -202,4 +218,25 @@ public CopySpec getMetaInf() {
     public CopySpec metaInf(Closure<?> configureClosure) {
         return ConfigureUtil.configure(configureClosure, getMetaInf());
     }
+
+    /**
+     * Provides a comparator that sorts the contents of a Jar/Ear/War so that META-INF is always first.
+     */
+    private static class JarContentsComparator implements Comparator<RelativePath> {
+        @Override
+        public int compare(RelativePath o1, RelativePath o2) {
+            boolean o1MetaInf = isMetaInf(o1.getSegments()[0]);
+            boolean o2MetaInf = isMetaInf(o2.getSegments()[0]);
+            if (o1MetaInf && !o2MetaInf) {
+                return -1;
+            } else if (!o1MetaInf && o2MetaInf) {
+                return 1;
+            }
+            return o1.compareTo(o2);
+        }
+
+        private boolean isMetaInf(String root) {
+            return root.equalsIgnoreCase("META-INF");
```

Comment:
Looks like it. I do not quite understand why. Probably for some case insensitive file systems one cannot enforce the case.